scene/realistic-lighting-and-shadows/src/main/java/com/esri/samples/realistic_lighting_and_shadows/RealisticLightingAndShadowsController.java

```diff
@@ -176,6 +172,43 @@ public Double fromString(String string) {
     });
   }
 
+  /**
+   * Set labels to display in the combo box.
+   */
+  private void setComboBoxLabels() {
+
+    comboBox.setConverter(new StringConverter<LightingMode>() {
+
+      @Override
+      public String toString(LightingMode mode) {
+
+        if (mode == LightingMode.LIGHT) return "Light Only";
+        if (mode == LightingMode.LIGHT_AND_SHADOWS) return "Light and Shadows";
+        if (mode == LightingMode.NO_LIGHT) return "No Light";
+        else return "Light only";
+      }
+
+      @Override
+      public LightingMode fromString(String string) {
+        return null;
+      }
+    });
+
+    comboBox.setCellFactory(comboBox -> new ListCell<LightingMode>() {
```

Comment:
Nice work on using the cell factory here!
We often put these new definitions of the ListCell class in their own file. Check out an example here:
https://github.com/Esri/arcgis-runtime-samples-java/blob/master/utility_network/trace-a-utility-network/src/main/java/com/esri/samples/trace_a_utility_network/UtilityTerminalListCell.java

With the if/else block you have below, we might be able to get away without the `StringConverter` above on lines 180- 195 (but I might be wrong). This is because you can shove the values from the `LightningMode` enum into the combobox, and use the ListCell to customise what is actually displayed in the UI.